grails-app/services/au/org/ala/ecodata/MetadataService.groovy

```diff
@@ -135,8 +135,34 @@ class MetadataService {
         // Remove deprecated activities
         activities = activities.findAll {!it.status || it.status == ACTIVE}
 
-        Map byCategory = [:]
+        groupActivities(activities)
+    }
+
+    /**
+     * Returns a Map of activity types grouped by category.
+     * @param programId Most of the legacy projects are updated recently to insert programId (see github #2741)
+     * @return a Map, key: String, value: List of name, description for each activity in the category
+     */
+    Map activitiesListByProgramId(String programId) {
+        Map config = Program.findByProgramId(programId)?.config
+        List<String> activities = config?.activities.collect{it.name}
```

Comment:
This will still throw an NPE if the config has no activities attached.  Please include in your "missing config" unit test a scenario includes a Program with a non-null config object with no "activities" key.

Reply:
i've added this test method, should this be ok?

<img width="1156" alt="Screenshot 2023-05-16 at 10 53 14 am" src="https://github.com/AtlasOfLivingAustralia/ecodata/assets/90952854/e537fbcf-23eb-437d-a939-17c11ebeb3c9">